### ISSUE TITLE
Select rows sends and shanges selection when it should not

### ISSUE THREAD

**Issue by Hrovatin** [biolab/orange3#4316]
Similar to https://github.com/biolab/orange3/issues/4278 that was already closed.
When I change group in widget marked by A the selection changes in B. This should not happen. What is even stranger, even if I disable automatic sending in B the data after B changes according to the change in B upon me changing A.
![Screenshot from 2020-01-09 10-22-56](https://user-images.githubusercontent.com/47607471/72056416-d961bd80-32cc-11ea-9f0f-8ed1b56e8ec7.png)

Orange 3.25.0 (last master, if I am not mistaken).

**Comment by ajdapretnar** [comment]
Would you be able to provide a simpler workflow to reproduce the issue?

**Comment by Hrovatin** [comment]
Simpler workflow: zoo.tab -> select rows -> select rows -> table

**Comment by ajdapretnar** [comment]
I could not reproduce this, until I updated orange-widget-base and orange-canvas-core to the new master. I suspect one of the changes in orange-widget-base broke the previous fixed issue.

**Comment by janezd** [comment]
I can reproduce it by enabling Remove unused features. What happenned was that the widget then retrieves one of the past contexts. In larger schema, even if Remove unused features is disabled, some of the widgets in between may change due to changed data and cause Select Rows to retrieve an old concept.

Increasing settings version should not help (as in #4319). Version was properly increased in the previous commit, and migrations are disabled anyway. It could be that the widget does not reuse settings from previous versions, though I'm not aware of any such checks.

As for sending data when auto commit is disabled: this works as intended. The check box is about applying changes, not blocking signals.

In summary: context settings may lead to weird situations like this one, but there is no way around it.
